### DISCUSSION TITLE
Custom Hedera Token Service Fees

### DISCUSSION THREAD

**Discussion by Cooper-Kunz** [hiero-ledger/hiero-improvement-proposals#90] [Ideas]
We propose adding a limited set of functionality to the Hedera Token Service (HTS) that allows developers to define custom fees between different tokens at time of creation, or at a later modification date with Hedera’s controlled mutability[1](https://hedera.com/blog/code-is-law-but-what-if-the-law-needs-to-change). These custom fees will enable dynamic ecosystem fees as we see with projects such as Uniswap[2](https://uniswap.org/docs/v2/protocol-overview/how-uniswap-works/), or even Hedera itself[3](https://hedera.com/hh_whitepaper_v2.1-20200815.pdf). Additional implementations could even allow implementation of custom, and if desired, perpetual royalties[4](https://messari.io/article/explain-it-like-i-am-5-nfts), for any token issued on HTS, whether fungible or non-fungible.

Update: this discussion has been moved to #92

**Comment by Cooper-Kunz** [comment]
The initial proposal can be found in #91 - would love to get the community's thoughts on this!

**Reply by Cooper-Kunz** [reply]
Moved to https://github.com/hashgraph/hedera-improvement-proposal/discussions/92

**Comment by bugbytesinc** [comment]
Thinking out loud: if one can modify an exchange schedule, should there be a time dimension added?  In other words, would it be beneficial to enforce "this exchange ratio changes on XXX" when the feature allowing changes is enabled?

**Reply by Cooper-Kunz** [reply]
Hey @bugbytesinc are you able to move this question to #92? 

Certainly an interesting suggestion - let's chat about it over there.
